programs/vest_api/src/vest_processor.rs

```diff
@@ -0,0 +1,342 @@
+//! vest program
+use crate::{
+    vest_instruction::{VestError, VestInstruction},
+    vest_state::VestState,
+};
+use bincode::deserialize;
+use chrono::prelude::*;
+use log::*;
+use solana_sdk::{account::KeyedAccount, instruction::InstructionError, pubkey::Pubkey};
+
+/// Return the date that is 'n' months from 'start'.
+fn get_month(start: &Date<Utc>, n: u32) -> Date<Utc> {
+    let year = start.year() + (start.month0() + n) as i32 / 12;
+    let month0 = (start.month0() + n) % 12;
+
+    // For those that started on the 31st, pay out on the latest day of the month.
+    let mut dt = None;
+    let mut days_back = 0;
+    while dt.is_none() {
+        dt = Utc
```

Comment:
you can't use DTC in programs, wallclocks differ box-to-box

in other places where time passes, we use sysvar::clock...  specify times in Slots

Reply:
The date comes from a trusted account, not from validators.

Reply:
you're supposed to compare the witness with the oracle?  why not vest in terms of slots?

Reply:
I'm worried someone will speed up the clock

Reply:
yeah, but would that be that bad?

Reply:
@aeyakovenko, thoughts on a date oracle versus cluster time?